iwyu_tool.py

```diff
@@ -91,6 +91,73 @@ def clang_formatter(output):
 }
 
 
+if sys.platform.startswith('win'):
+    # Case-insensitive match on Windows
+    def normcase(s):
+        return s.lower()
+else:
+    def normcase(s):
+        return s
+
+
+def is_msvc_driver(compile_command):
+    """ Return True if compile_command matches an MSVC CL-style driver. """
+    compile_command = normcase(compile_command)
+
+    if compile_command.endswith('cl.exe'):
+        # Native MSVC compiler or clang-cl.exe
+        return True
+
+    if compile_command.endswith('clang-cl'):
+        # Cross clang-cl on non-Windows
+        return True
+
+    return False
+
+
+def is_subpath_of(path, parent):
+    """ Return True if path is equal to or fully contained within parent.
+
+    Assumes both paths are canonicalized with os.path.realpath.
+    """
+    parent = normcase(parent)
+    path = normcase(path)
+
+    if path == parent:
+        return True
+
+    if not path.startswith(parent):
+        return False
+
+    # Now we know parent is a prefix of path, but they only share lineage if the
+    # difference between them starts with a path separator, e.g. /a/b/c/file
+    # is not a parent of /a/b/c/file.cpp, but /a/b/c and /a/b/c/ are.
+    parent = parent.rstrip(os.path.sep)
+    suffix = path[len(parent):]
+    return suffix.startswith(os.path.sep)
+
+
+def split_command(cmdstr):
+    """ Split a command string into a list, respecting shell quoting. """
+    def strip_quotes(arg):
+        if arg.startswith('"') and arg.endswith('"'):
+            return arg[1:-1]
+        return arg
+
+    # shlex.split with posix=False is known to not work perfectly for Windows
+    # command-lines, but it seems good enough in the general case for
+    # compilation databases.
+    windows = sys.platform.startswith('win')
+    cmd = shlex.split(cmdstr, posix=not windows)
+
+    # It does, however, retain quotes around arguments, which causes
+    # double-quoting and escaping when passed on to subprocess. Strip them.
+    if windows:
+        cmd = [strip_quotes(a) for a in cmd]
```

Comment:
I don't think it does break. I looked through subprocess.py and saw that on Windows, the first thing that happens is that the `args` list is passed to `list2cmdline`, which joins the arguments back to a string and quotes them if necessary. As part of that quoting, though, it escapes any existing quotes.

So an argument like `"C:\my file\with spaces.txt"` would turn into `"\"C:\my file\with spaces.txt\""`, which shouldn't be found.

I think it might make sense to add some of this rationale to the comment, though, I see now that I didn't mention subprocess there.

Reply:
I think I know what's going on now. The problem is not in the strip_quotes() function, but in shlex.split(.., posix=False). It splits incorrectly when a quoted argument with a space is appended to an unquoted one (|**something"with spaces"**| becomes |**something"with**|**spaces**|):

To verify this, I added an include directory with a space to the IWYU's root path and and to CMakeLists.txt:
```cmake
target_include_directories(include-what-you-use PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/inc\ ludes")
```

After this, the generation step will add this include path for each command in the compile_commands.json file:
```
-I"..\\tools\\clang\\tools\\iwyu\\inc ludes"  
```

When I run it under iwyu_tool.py I get this as the first line of output:
```
error: no such file or directory: 'ludes"'
```

So, in short, directories with spaces will not work on Windows. :(